Cython/Utility/Synchronization.c

```diff
@@ -359,6 +370,26 @@ static void __Pyx__Locks_PyThreadTypeLock_Lock(__Pyx_Locks_PyThreadTypeLock lock
 #define __Pyx_Locks_PyMutex_LockGil(l) PyMutex_Lock(&l)
 #define  __Pyx_Locks_PyMutex_LockNogil(l) PyMutex_Lock(&l)
 
+// For Python 3.13, we need to check if PyMutex_IsLocked is available
+#if PY_VERSION_HEX >= 0x030e00C1
+// Python 3.14+ has PyMutex_IsLocked  
+static CYTHON_INLINE int __Pyx_Locks_PyMutex_Locked(__Pyx_Locks_PyMutex lock) {
```

Comment:
This would probably be better as a define.

You're passing `lock` as a value here so the read won't happen atomically. A define avoids that